crates/ra_parser/src/grammar/items.rs

```diff
@@ -118,7 +118,15 @@ pub(super) fn maybe_item(p: &mut Parser, m: Marker, flavor: ItemFlavor) -> Resul
         && p.at_contextual_kw("default")
         && (match p.nth(1) {
             T![impl] => true,
-            T![fn] | T![type] => {
+            T![unsafe] => {
+                if T![impl] == p.nth(2) {
+                    p.bump_remap(T![default]);
+                    p.bump_remap(T![unsafe]);
```

Comment:
```suggestion
                    p.bump(T![unsafe]);
```

no need to remap unsafe, it's not a contextual keyword.